packages/safe-apps-sdk/src/wallet/wallet.test.ts

```diff
@@ -0,0 +1,89 @@
+import SDK from '../sdk';
+import { Methods } from '../communication/methods';
+import { PostMessageOptions } from '../types';
+import { Permission } from '../types/permissions';
+
+describe('Safe Apps SDK wallet methods', () => {
+  const sdkInstance = new SDK();
+  let postMessageSpy: jest.SpyInstance<void, [message: any, options?: PostMessageOptions]>;
+
+  beforeEach(() => {
+    postMessageSpy = jest.spyOn(window.parent, 'postMessage');
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('SDK.wallet.getPermissions', () => {
+    test('Should send a valid message to the interface', () => {
+      sdkInstance.wallet.getPermissions();
+
+      expect(postMessageSpy).toHaveBeenCalledWith(
+        expect.objectContaining({ method: Methods.wallet_getPermissions, params: undefined }),
+        '*',
+      );
+    });
+
+    test('should resolve the correct getPermissions types', async () => {
```

Comment:
This test is obsolete. It checks if the value equals mocked object, it only tests jest's mocking capabilities

Reply:
If you want to test how it processes the response (this is an excellent idea btw), you can send the response via `window.postMessage`, so it goes through the whole message flow (received by the communicator and then passed back to the calling method)

Reply:
This is the entypoint for the message: https://github.com/safe-global/safe-apps-sdk/blob/master/packages/safe-apps-sdk/src/communication/index.ts#L19

Reply:
Totally right 🙈

Changed